tests/test_constraints_discrete.py

```diff
@@ -238,3 +245,39 @@ def test_custom(campaign):
         & campaign.searchspace.discrete.exp_rep["Solvent_1"].eq("C3")
     ).sum()
     assert num_entries == 0
+
+
+@pytest.mark.parametrize(
+    "parameter_names",
+    [
+        [
+            "Temperature",
+            "Solvent_1",
+            "Solvent_2",
+            "Solvent_3",
+            "Fraction_1",
+            "Fraction_2",
+            "Fraction_3",
+        ]
+    ],
+)
+@pytest.mark.parametrize(
+    "constraint_names",
+    [
+        ["Constraint_4"],
+        ["Constraint_12"],
+        ["Constraint_4", "Constraint_8", "Constraint_9"],
+    ],
+)
+def test_polars_product(constraints, parameters):
```

Comment:
while I have nothing against this test we had others in mind too that test the application of filtering via polars versus pandas for each for the three constraint types that have it
imo not needed for this PR but we would possibly extend that to having it tested with a wide range of constraints of these types via hypothesis